aten/src/ATen/function_wrapper.py

```diff
@@ -171,6 +185,21 @@ def TypedDict(name, attrs, total=True):  # type: ignore
 }
 """)
 
+C10_FUNCTION_DEFINITION = CodeTemplate("""\
+static inline ${return_type} ${api_name}(${formals}) {
+#ifdef USE_STATIC_DISPATCH
+    ${mobile_function_body}
+#else
+    static c10::OperatorHandle op = c10::Dispatcher::singleton().findSchema({"aten::${name}", "${overload_name}"}).value();
+    if (${inferred_is_variable}) {
+        return c10::Dispatcher::singleton().callUnboxedAutogradKernel<${formals_types_with_return}>(op ${native_actuals_with_comma_prefix});
```

Comment:
It seems pretty weird to me that I have to do another singleton lookup to actually call an OperatorHandle. Shouldn't the OperatorHandle know what dispatcher it's associated with?

Reply:
The reason is that the Dispatcher class is a facade API. All calls to the dispatcher go through the Dispatcher class currently, so it's all in one place. The `OperatorHandle` class is just a handle (basically an operator ID) without any functionality. We can consider changing that later and giving the `OperatorHandle` class more responsibility, but that's not how the API is currently implemented. In any case, probably not something we should change in this diff since that would be a bigger and unrelated API change to the dispatcher.